src/EoS/Nuclear/CPU_EoS_Nuclear.cpp

```diff
@@ -454,18 +481,266 @@ static real EoS_DensPres2CSqr_Nuclear( const real Dens_Code, const real Pres_Cod
       printf( "        Passive scalars:" );
       for (int v=0; v<NCOMP_PASSIVE; v++)    printf( " %d=%13.7e", v, Passive_Code[v] );
       printf( "\n" );
-      printf( "        EoS error code: %d\n", Err );
 #     endif
+      printf( "        EoS error code: %d\n", Err );
    }
 #  endif // GAMER_DEBUG
 
 
+// store extra output variables if required
+   /*
+   if ( ExtraInOut != NULL )
+   {
+      ExtraInOut[...] = ...;
+   }
+   */
+
+
    return Cs2_Code;
 
 } // FUNCTION : EoS_DensPres2CSqr_Nuclear
 
 
 
+//-------------------------------------------------------------------------------------------------------
+// Function    :  EoS_General_Nuclear
+// Description :  General EoS converter: In[] -> Out[]
+//
+// Note        :  1. See EoS_DensEint2Pres_Nuclear()
+//                2. In[] and Out[] must NOT overlap
+//                3. Support the temperature and entropy modes:
+//                   (1) Temperature mode:
+//                         In [0] = mass density in code units
+//                         In [1] = temperature in kelvin
+//                         In [2] = Ye
+//                         Out[0] = volume energy density in code units (with energy shift)
+//                         Out[1] = entropy in kB/baryon
+//                   (2) Entropy mode:
+//                         In [0] = mass density in code units
+//                         In [1] = entropy in kB/baryon
+//                         In [2] = Ye
+//                         Out[0] = volume energy density in code units (with energy shift)
+//                   --> Both In[] and Out[] can be extended if necessary
+//                4. No units conversion will be applied to entropy and Ye
+//
+// Parameter   :  Mode       : NUC_MODE_TEMP or NUC_MODE_ENTR
+//                Out        : Output array     (see the Note above)
+//                In         : Input array      (see the Note above)
+//                AuxArray_* : Auxiliary arrays (see the Note above)
+//                Table      : EoS tables
+//
+// Return      :  Out[]
+//-------------------------------------------------------------------------------------------------------
+GPU_DEVICE_NOINLINE
+static void EoS_General_Nuclear( const int Mode, real Out[], const real In[], const double AuxArray_Flt[],
```

Comment:
It seems that this function can be run on GPU. Just curious about where the size of `In` and `Out` is defined and the function for transferring data locates?

Reply:
@hfhsieh `In[]` and `Out[]` are just pointers. So we do not need to specify their sizes and transfer data.